packages/core/strapi/lib/services/entity-service/types/params/populate.d.ts

```diff
@@ -0,0 +1,39 @@
+import type { Attribute, Common, Utils } from '@strapi/strapi';
+
+/**
+ * Wildcard notation for populate
+ *
+ * To populate all the root level relations
+ */
+type WildcardNotation = '*';
+
+/**
+ * Union of all possible string representation for populate
+ *
+ * @example
+ * type A = 'image'; // ✅
+ * type B = 'image,component'; // ✅
+ * type c = '*'; // ✅
+ * type D = 'populatableField'; // ✅
+ * type E = '<random_string>'; // ❌
+ */
+type StringNotation<TSchemaUID extends Common.UID.Schema> =
+  | WildcardNotation
+  | Attribute.GetPopulatableKeys<TSchemaUID>
+  | `${string},${string}`;
+
+/**
+ * Array notation for populate
+ *
+ * @example
+ * type A = ['image']; // ✅
+ * type B = ['image', 'component']; // ✅
+ * type C = ['populatableField']; // ✅
+ * type D = ['<random_string>']; // ❌
+ */
+type ArrayNotation<TSchemaUID extends Common.UID.Schema> =
+  Attribute.GetPopulatableKeys<TSchemaUID>[];
+
+export type Any<TSchemaUID extends Common.UID.Schema> =
```

Comment:
Are we adding advanced populate types in a later PR but before the target PR is merged? It feels like having something like `| Object` would work as an essentially untyped placeholder unti we get around to implementing it.

Reply:
yeah it is the next thing am working on, yeah I could add that

Reply:
Since nothing will be released before adding the object notation, I'm not sure it adds value. But no strong opinion either.